core/shared/src/main/scala/scalaz/zio/Schedule.scala

```diff
@@ -586,8 +586,13 @@ object Schedule extends Serializable {
   /**
    * A schedule that recurs the specified number of times. Returns the number
    * of repetitions so far.
+   *
+   * If 0 of negative numbers are given, the operation is not done at all so
+   * that in (op: IO[E, A]).repeat(Schedule.recurs(0)) , op is not done at all.
    */
-  final def recurs(n: Int): Schedule[Any, Int] = forever.whileOutput(_ < n)
+  final def recurs(n: Int): Schedule[Any, Int] =
+    if(n < 1) Schedule[Int, Any, Int](IO.point(0), (_, _) => IO.now(Decision.done(Duration.Zero, 0, 0)))
```

Comment:
This construction seems to totally avoid the execution of the repeated `IO`. @jdegoes believes it should not be the case and want to take a look at that.

Reply:
I found the explanation: `io.repeat` will return the `B` in the `Schedule`, which in this case, is always exactly equal to 0.

To fix the tests, change this:

```scala
s <- ref.update(_ + 1).repeat(Schedule.recurs(n))
``` 

to this:

```scala
s <- ref.update(_ + 1).repeat(Schedule.recurs(n)) *> ref.get
``` 

That way you will get the latest value from the `ref`, rather than getting the output of the `Schedule`.

Reply:
oh yes, of course :/